### PR TITLE
Functional Tests - Enable/Disable shop

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Test enable Disable/Enable shop
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | TEST_PATH="functional/BO/shopParameters/general/maintenance/01_enableDisableShop.js" URL_FO=yourShopURL npm run specific-test

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/17029)

### CODE REVIEW

File: tests/puppeteer/pages/FO/FObasePage.js [hunk 1]
```diff
@@ -133,4 +134,12 @@ module.exports = class Home extends CommonPage {
   async getFooterLinksBlockTitle(position) {
     return this.getTextContent(this.wrapperTitle.replace('%POSITION', position));
   }
+
+  /**
+   * Get Page content
+   * @return {Promise<textContent>}
+   */
+  getPageContent() {
```
Comment (SimonGrn) [inline tests/puppeteer/pages/FO/FObasePage.js]: I guess this is the same kind of method than in `BoBasePage` ?

Comment (nesrineabdmouleh) [reply tests/puppeteer/pages/FO/FObasePage.js]: Yes it's the same method in `commonPage` -> `getTextContent`
I will change it :)

Comment (SimonGrn) [reply tests/puppeteer/pages/FO/FObasePage.js]: Maybe we can move it to `BasePage` ? (Or `CommonPage` I don't remember the parent)

File: tests/puppeteer/campaigns/functional/BO/shopParameters/general/maintenance/01_enableDisableShop.js [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+const BOBasePage = require('@pages/BO/BObasePage');
+const LoginPage = require('@pages/BO/login');
+const DashboardPage = require('@pages/BO/dashboard');
+const GeneralPage = require('@pages/BO/shopParameters/general');
+const MaintenancePage = require('@pages/BO/shopParameters/general/maintenance');
+const HomePage = require('@pages/FO/home');
+
+let browser;
+let page;
+const maintenanceText = 'We are currently updating our shop and will be back really soon. Thanks for your patience.';
```
Comment (PierreRambaud) [inline tests/puppeteer/campaigns/functional/BO/shopParameters/general/maintenance/01_enableDisableShop.js]: Should be in fixture file?

Comment (PierreRambaud) [inline tests/puppeteer/campaigns/functional/BO/shopParameters/general/maintenance/01_enableDisableShop.js]: Should be in MaintenancePage no?

File: tests/puppeteer/pages/BO/shopParameters/general/maintenance/index.js [hunk 3]
```diff
@@ -29,12 +32,44 @@ module.exports = class shopParamsMaintenance extends BOBasePage {
...
+   */
+  async addMyIpAddress() {
+    await this.page.click(this.addMyIPAddressButton);
+    await this.page.click(this.saveFormButton);
+    return this.getTextContent(this.alertSuccessBloc);
+  }
+
+  /**
+   * Add maintenance IP address input
+   * @ipAddress
```
Comment (PierreRambaud) [inline tests/puppeteer/pages/BO/shopParameters/general/maintenance/index.js]:
```suggestion
   * @param ipAddress
```

File: tests/puppeteer/pages/BO/shopParameters/general/maintenance/index.js [hunk 4]
```diff
@@ -29,12 +32,44 @@ module.exports = class shopParamsMaintenance extends BOBasePage {
...
     else await this.page.click(this.disableShopLabel);
     await this.page.click(this.saveFormButton);
-    await this.page.waitForSelector(this.alertSuccessBloc);
+    return this.getTextContent(this.alertSuccessBloc);
+  }
+
+  /**
+   * Update Maintenance text
+   * @param text
+   * @return {Promise<textContent>}
```
Comment (PierreRambaud) [inline tests/puppeteer/pages/BO/shopParameters/general/maintenance/index.js]: The docblock for `getTextContent` function is wrong.
it must be `@return string` and not `@return textContent` which is not a js type. (in commonPage.js)

```suggestion
   * @return {Promise<string>}
```

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
Docblock need to be changed

File: tests/puppeteer/campaigns/functional/BO/shopParameters/general/maintenance/01_enableDisableShop.js [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+const BOBasePage = require('@pages/BO/BObasePage');
+const LoginPage = require('@pages/BO/login');
+const DashboardPage = require('@pages/BO/dashboard');
+const GeneralPage = require('@pages/BO/shopParameters/general');
+const MaintenancePage = require('@pages/BO/shopParameters/general/maintenance');
+const HomePage = require('@pages/FO/home');
+
+let browser;
+let page;
+const newMaintenanceText = 'Maintenance';
```
Comment (PierreRambaud) [inline tests/puppeteer/campaigns/functional/BO/shopParameters/general/maintenance/01_enableDisableShop.js:16]: This one too :sweat_smile:

File: tests/puppeteer/pages/BO/shopParameters/general/maintenance/index.js [hunk 6]
```diff
@@ -29,12 +33,44 @@ module.exports = class shopParamsMaintenance extends BOBasePage {
...
     else await this.page.click(this.disableShopLabel);
     await this.page.click(this.saveFormButton);
-    await this.page.waitForSelector(this.alertSuccessBloc);
+    return this.getTextContent(this.alertSuccessBloc);
+  }
+
+  /**
+   * Update Maintenance text
+   * @param text
+   * @return {Promise<string>}
```
Comment (PierreRambaud) [inline tests/puppeteer/pages/BO/shopParameters/general/maintenance/index.js:48]: Could you update the original getTextContent dock block please?

File: tests/puppeteer/campaigns/functional/BO/shopParameters/general/maintenance/01_enableDisableShop.js (hunk 5 above)
Comment (nesrineabdmouleh) [reply tests/puppeteer/campaigns/functional/BO/shopParameters/general/maintenance/01_enableDisableShop.js:16]: I use this one only in this test, then I think that it should be in this file not on the page.

**Review by PierreRambaud — APPROVED** [review]

**Pull request merged**